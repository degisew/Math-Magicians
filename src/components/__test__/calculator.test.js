/* eslint-disable */
import { render, screen, fireEvent } from '@testing-library/react';
import renderer from 'react-test-renderer';
import Calculator from '../Calculator';

it('Did component changed', () => {
  const tree = renderer.create(
    <Calculator />,
  ).toJSON();
  expect(tree).toMatchSnapshot();
});

it('should throw 6  for (1+5)', () => {
  render(
    <Calculator />,
  );
  const display = screen.getByTestId('display');
  const one = screen.getByText('1');
  const five = screen.getByText('5');
  const plus = screen.getByText('+');
  const equal = screen.getByText('=');
  fireEvent.click(one);
  fireEvent.click(plus);
  fireEvent.click(five);
  fireEvent.click(equal);
  expect(display.textContent).toBe('6');
});
