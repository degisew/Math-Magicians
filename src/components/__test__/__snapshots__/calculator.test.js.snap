// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Did component changed 1`] = `
<div
  className="calc-container"
>
  <div
    className="calc-screen"
    data-testid="display"
  >
    <span>
      0
    </span>
    <span />
    <span />
  </div>
  <div
    className="td row1-col1"
  >
    <button
      className="cal-button"
      data-testid="button"
      onClick={[Function]}
    >
      AC
    </button>
  </div>
  <div
    className="td row1-col2"
  >
    <button
      className="cal-button"
      data-testid="button"
      onClick={[Function]}
    >
      +/-
    </button>
  </div>
  <div
    className="td row1-col3"
  >
    <button
      className="cal-button"
      data-testid="button"
      onClick={[Function]}
    >
      %
    </button>
  </div>
  <div
    className="td row1-col4 operator"
  >
    <button
      className="cal-button"
      data-testid="button"
      onClick={[Function]}
    >
      รท
    </button>
  </div>
  <div
    className="td row2-col1"
  >
    <button
      className="cal-button"
      data-testid="button"
      onClick={[Function]}
    >
      7
    </button>
  </div>
  <div
    className="td row2-col2"
  >
    <button
      className="cal-button"
      data-testid="button"
      onClick={[Function]}
    >
      8
    </button>
  </div>
  <div
    className="td row2-col3"
  >
    <button
      className="cal-button"
      data-testid="button"
      onClick={[Function]}
    >
      9
    </button>
  </div>
  <div
    className="td row2-col4 operator"
  >
    <button
      className="cal-button"
      data-testid="button"
      onClick={[Function]}
    >
      x
    </button>
  </div>
  <div
    className="td row3-col1"
  >
    <button
      className="cal-button"
      data-testid="button"
      onClick={[Function]}
    >
      4
    </button>
  </div>
  <div
    className="td row3-col2"
  >
    <button
      className="cal-button"
      data-testid="button"
      onClick={[Function]}
    >
      5
    </button>
  </div>
  <div
    className="td row3-col3"
  >
    <button
      className="cal-button"
      data-testid="button"
      onClick={[Function]}
    >
      6
    </button>
  </div>
  <div
    className="td row3-col4 operator"
  >
    <button
      className="cal-button"
      data-testid="button"
      onClick={[Function]}
    >
      -
    </button>
  </div>
  <div
    className="td row4-col1"
  >
    <button
      className="cal-button"
      data-testid="button"
      onClick={[Function]}
    >
      1
    </button>
  </div>
  <div
    className="td row4-col2"
  >
    <button
      className="cal-button"
      data-testid="button"
      onClick={[Function]}
    >
      2
    </button>
  </div>
  <div
    className="td row4-col3"
  >
    <button
      className="cal-button"
      data-testid="button"
      onClick={[Function]}
    >
      3
    </button>
  </div>
  <div
    className="td row4-col4 operator"
  >
    <button
      className="cal-button"
      data-testid="button"
      onClick={[Function]}
    >
      +
    </button>
  </div>
  <div
    className="td row5-col1"
  >
    <button
      className="cal-button"
      data-testid="button"
      onClick={[Function]}
    >
      0
    </button>
  </div>
  <div
    className="td row5-col2"
  >
    <button
      className="cal-button"
      data-testid="button"
      onClick={[Function]}
    >
      .
    </button>
  </div>
  <div
    className="td row5-col3 operator"
  >
    <button
      className="cal-button"
      data-testid="button"
      onClick={[Function]}
    >
      =
    </button>
  </div>
</div>
`;
